src/main/java/biz/paluch/spinach/output/JobOutput.java

```diff
@@ -35,13 +39,24 @@ public void set(ByteBuffer bytes) {
             return;
         }
 
-        body = codec.decodeValue(bytes);
+        if (body == null) {
+            counters = new HashMap<String, Long>();
```

Comment:
Initializing the `counters` map eagerly (e. g. field level) prevents a null-map. WDYT?